src/tests/stats/test_distribution_uniform.fypp

```diff
@@ -0,0 +1,298 @@
+        complex, parameter :: za(10) = [(8.0, 8.0), (7.0, 7.0), (4.0, 4.0),    &
+                                        (1.0, 1.0), (5.0, 5.0), (9.0, 9.0),    &
+                                        (6.0, 6.0), (3.0, 3.0), (2.0, 2.0),    &
+                                        (10.0, 10.0)]
+        integer :: i, put, get
+
+        do i = 1, 10
+            n(i) = i
+            x(i) = real(i)
+            z(i) = cmplx(real(i),real(i))
+        end do
+        put = 32165498
+        call random_seed(put, get)
+        n(:) = shuffle(n)
+        x(:) = shuffle(x)
+        z(:) = shuffle(z)
+        call check(all(n == na),                                               &
+            msg = "Integer shuffle failed test", warn=warn)
+        call check(all(x == xa),                                               &
+            msg = "Real shuffle failed test", warn=warn)
+        call check(all(z == za),                                               &
+            msg = "Complex shuffle failed test", warn=warn)
+    end subroutine test_shuffle
+
+
+
+    subroutine test_uni_rvs_0
+        integer, parameter :: num = 10000000
+        integer, parameter :: array_size = 1000
+        integer :: i, j, freq(0 : array_size - 1)
+        real(dp) :: chisq, expct
+
+        print *,""
+        print *, "Test uniform random generator with chi-squared"
+        freq = 0
+        do i = 1, num
+            j = array_size * uni_rvs( )
+            freq(j) = freq(j) + 1
+        end do
+        chisq = 0.0_dp
+        expct = num / array_size
+        do i = 0, array_size - 1
+           chisq = chisq + (freq(i) - expct) ** 2 / expct
+        end do
+        write(*,*) "The critical value for chi-squared with 1000 d. of f. is"  &
+            //" 1143.92"
+        write(*,*) "Chi-squared for uniform random generator is : ", chisq
+        call check((chisq < 1143.9) ,                                          &
+            msg = "uniform randomness failed chi-squared test", warn=warn)
+    end subroutine test_uni_rvs_0
+
+
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+      subroutine test_uni_rvs_${t1[0]}$${k1}$
+          ${t1}$ :: res(15), scale, loc
+          integer :: i, n, seed, get, k
+
+          #:if k1 == "int8"
+          ${t1}$, parameter :: ans(15) = [47, 99, 43, 37, 48, 30, 27, 100, 30, &
+                                          33, 21, 103, 55, 54, 110]
+          #:elif k1 == "int16"
+          ${t1}$, parameter :: ans(15) = [25, 4, 81, 98, 49, 34, 32, 62, 115,  &
+                                          112, 26, 20, 37, 100, 82]
+          #:elif k1 == "int32"
+          ${t1}$, parameter :: ans(15) = [19, 52, 56, 20, 59, 44, 34, 102, 19, &
+                                          39, 60, 50, 97, 56, 67]
+          #:elif k1 == "int64"
+          ${t1}$, parameter :: ans(15) = [76, 45, 43, 75, 76, 15, 25, 24, 114, &
+                                          113, 94, 29, 109, 93, 89]
+          #:elif t1[0] == "r"
```

Comment:
Minor point -- here I would suggest matching `real` rather than `r`, simply for clarity -- something like `#:elif t1[0:4] == 'real'` (assuming I got the python string indexing correct).

Reply:
Will leave comment on the line.